### PR TITLE
Continuing tunneling support for the proxy...

### PR DESCRIPTION
**Pull request by brendandburns**
Builds off of #8898 

@cjcullen @roberthbailey @lavalamp

Not the prettiest, but still a good buy.

### CODE REVIEW

**Comment by brendandburns** [comment]
(I need to plumb this through the unit tests too)

**Comment by bgrant0607** [comment]
cc @ncdc

**Comment by ncdc** [comment]
We'll potentially have issues shipping go-crypto w/o export control approval.

cc @vbatts @smarterclayton

**Comment by smarterclayton** [comment]
Yeah I'm not sure we can enable this by default in our packages.  Can we decouple this more so it can be compiled out by a build tag?

**Comment by smarterclayton** [comment]
Also, can we move this out of pkg/util/ and into pkg/util/ssh?  Util should be kept small - I don't need go crypto to run util.Forever

**Comment by smarterclayton** [comment]
Why is this required for 1.0?

**Comment by vbatts** [comment]
not sure about for EL, but fedora is already there (http://koji.fedoraproject.org/koji/buildinfo?buildID=605510)

**Comment by brendandburns** [comment]
@smarterclayton this is necessary to make the proxy work for Google Container Engine when we are hosting the master in a different project that we own and manage.

fwiw, this isn't adding any new dependencies, they were already in the repo.

**Comment by smarterclayton** [comment]
Ok - I'm not necessarily opposed, just didn't know the context.  Vincent, are we ok on the crypto for this?

**Comment by brendandburns** [comment]
@smarterclayton if you want me to hide this behind a build flag, let me know.  It wouldn't be hard to do.

I'm also happy to split it out into a separate directory.

**Comment by vbatts** [comment]
I'll have to double check if that has been filed yet

**Comment by smarterclayton** [comment]
Separate directory would make it easier if we do run into trouble with export control.

**Comment by brendandburns** [comment]
@smarterclayton moved it out to a separate directory.  ptal.

Thanks!
--brendan

File: pkg/util/ssh/ssh.go [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+	}
+	signer, err := ssh.ParsePrivateKey(buffer)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing SSH key %s: '%v'", key, err)
+	}
+	return signer, nil
+}
+
+/*
+if len(r.tunnels) == 0 {
```
Comment (bgrant0607) [inline pkg/util/ssh/ssh.go]: Why is this commented out?

**Comment by bgrant0607** [comment]
cc @lavalamp

**Pull request closed without merging**